lib/inspec/resources/users.rb

```diff
@@ -314,6 +336,59 @@ def credentials
 
       @cred_cache = @user_provider.credentials(@username) unless @user_provider.nil?
     end
+
+    # helper method for has_authorized_key matcher
+    # checks for key is present as part of authorized keys on the server
+    def authorized_key(key)
+      # cat is used in unix system to display content of file; similarly type is used for windows
+      inspec.os.windows? ? bin = "type" : bin = "cat"
```

Comment:
you can do something like this bin = inspec.os.windows? ? "type" : "cat"

Reply:
Made the changes accordingly.